### PR TITLE
VR-8224 Write class docstring for HDFSPath blob

### PR DESCRIPTION
**Pull request by nhatsmrt**
It's been awhile since I last wrote docstrings... Could you double check that I'm using the right format?
Also I based the parameters type and description on test files + documentation of those classes, so lmk if I misunderstood any.

### CODE REVIEW

File: client/verta/verta/dataset/_hdfs.py [hunk 1]
```diff
@@ -13,7 +13,20 @@
...
+class HDFSPath(Path):
+    """
+    Captures metadata about files from HDFS.
+
+    Parameters
+    ----------
+    hdfs_client : hdfs.client.Client
+        Instance of HDFS client.
+    paths : str or list of strs
+        A path, or a list of paths, to files from HDFS.
```
Comment (convoliution) [inline client/verta/verta/dataset/_hdfs.py]: List is preferred!
```suggestion
    paths : list of strs
        List of paths to files from HDFS.
```

File: client/verta/verta/dataset/_hdfs.py [hunk 2]
```diff
@@ -69,6 +82,22 @@ def _hash_file(self, filepath):
...
     def with_spark(sc, paths):
+        """
+        Creates an HDFSPath blob with a SparkContext instance.
+
+        Parameters
+        ----------
+        sc : pyspark.SparkContext
+            SparkContext instance.
+        paths : str or list of strs
+            Path, or list of paths, to binary input data file(s) from HDFS.
```
Comment (convoliution) [inline client/verta/verta/dataset/_hdfs.py]: List is preferred!
```suggestion
        paths : list of strs
            List of paths to binary input data file(s) from HDFS.
```

**Review by convoliution — CHANGES_REQUESTED** [review]
Minor suggestions, otherwise these sound about right to me.

Could you also do me a huge, huge favor and add an entry to the dataset blob section of the [docs page](https://github.com/VertaAI/modeldb/blob/master/client/verta/docs/api/api/versioning.rst) as well?

File: client/verta/verta/dataset/_hdfs.py (hunk 1 above)
Comment (nhatsmrt) [reply client/verta/verta/dataset/_hdfs.py]: Fixed.

File: client/verta/verta/dataset/_hdfs.py (hunk 2 above)
Comment (nhatsmrt) [reply client/verta/verta/dataset/_hdfs.py]: Fixed.

**Comment by nhatsmrt** [comment]
@convoliution a new entry is added to the docs page! See latest commit.

**Review by convoliution — APPROVED** [review]
Thanks!

**Pull request merged**